### PR TITLE
Add Subobjects: classes, metadata, serialization, reflection, disasm

### PR DESCRIPTION
**Pull request by tex3d**

### CODE REVIEW

File: tools/clang/tools/dxcompiler/dxcdisassembler.cpp [hunk 1]
```diff
@@ -575,6 +576,143 @@ void PrintViewIdState(DxilModule &M, raw_string_ostream &OS,
...
+      uint32_t Flags = 0;
+      if (!obj.GetStateObjectConfig(Flags)) {
+        OS << "<error getting subobject>";
+        break;
+      }
+      PrintFlags<DXIL::StateObjectFlags>(OS, Flags);
+      break;
+    }
+    case DXIL::SubobjectKind::LocalRootSignature:
+      bLocalRS = true;
```
Comment (hekota) [inline tools/clang/tools/dxcompiler/dxcdisassembler.cpp]: __fallthrough;

File: tools/clang/tools/dxcompiler/dxcdisassembler.cpp [hunk 2]
```diff
@@ -575,6 +576,143 @@ void PrintViewIdState(DxilModule &M, raw_string_ostream &OS,
...
+    case DXIL::SubobjectKind::RaytracingShaderConfig: {
+      uint32_t MaxPayloadSizeInBytes;
+      uint32_t MaxAttributeSizeInBytes;
+      if (!obj.GetRaytracingShaderConfig(MaxPayloadSizeInBytes,
+                                         MaxAttributeSizeInBytes)) {
+        OS << "<error getting subobject>";
+        break;
+      }
+      OS << "MaxPayloadSizeInBytes = " << MaxPayloadSizeInBytes
+         << "MaxAttributeSizeInBytes = " << MaxAttributeSizeInBytes;
```
Comment (hekota) [inline tools/clang/tools/dxcompiler/dxcdisassembler.cpp]: add comma - ", MaxAttributeSizeInBytes = "

File: tools/clang/tools/dxcompiler/dxcdisassembler.cpp [hunk 3]
```diff
@@ -1289,6 +1427,61 @@ void PrintPipelineStateValidationRuntimeInfo(const char *pBuffer,
...
+    auto reader = pSubobjectTableReader->GetItem(i);
+    DXIL::SubobjectKind kind = reader.GetKind();
+    bool bLocalRS = false;
+    switch (kind) {
+    case DXIL::SubobjectKind::StateObjectConfig:
+      subobjects.CreateStateObjectConfig(reader.GetName(),
+        reader.GetStateObjectConfig_Flags());
+      break;
+    case DXIL::SubobjectKind::LocalRootSignature:
+      bLocalRS = true;
```
Comment (hekota) [inline tools/clang/tools/dxcompiler/dxcdisassembler.cpp]: __fallthrough;

**Comment by ehsannas** [comment]
This change seems to break the Linux build. Please consider patching it if the fix is trivial. Thanks!

**Comment by antiagainst** [comment]
Travis CI is happy now.

**Comment by tex3d** [comment]
Thanks!

**Pull request merged**